DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc

```diff
@@ -0,0 +1,1141 @@
+    unsigned int daughter = (*vtxs)[i].tracksSize();
+    if( daughter > (*vtxs)[greatestvtx].tracksSize()) greatestvtx = i;
+  }
+
+  if(nVertex<=0){
+    vtx =  reco::Vertex::Point(0,0,0);
+  }
+  vtx = (*vtxs)[greatestvtx].position();
+
+  int nGoodVertices = 0;
+
+  if (pvHandle.isValid())
+    {
+      for (unsigned i=0; i<pvHandle->size(); i++)
+	{
+	  if ((*pvHandle)[i].ndof() > 4 &&
+	      (fabs((*pvHandle)[i].z()) <= 24) &&
+	      (fabs((*pvHandle)[i].position().rho()) <= 2))
+	    nGoodVertices++;
+	}
+    }
+
+  mNvtx->Fill(nGoodVertices);
+
+
+  // Get the Jet collection
+  //----------------------------------------------------------------------------
+  
+  std::vector<Jet> recoJets;
+  recoJets.clear();
+  
+  edm::Handle<CaloJetCollection>  caloJets;
+  edm::Handle<JPTJetCollection>   jptJets;
+  edm::Handle<PFJetCollection>    pfJets;
+  edm::Handle<BasicJetCollection> basicJets;
+  
+  // Get the Particle flow candidates and the Voronoi variables 
+  edm::Handle<reco::PFCandidateCollection> pfCandidates; 
+  edm::Handle<CaloTowerCollection> caloCandidates; 
+  edm::Handle<reco::CandidateView> pfcandidates_;
+  edm::Handle<reco::CandidateView> calocandidates_;
+  
+  edm::Handle<edm::ValueMap<VoronoiBackground>> VsBackgrounds;
+  edm::Handle<std::vector<float>> vn_;
+
+  edm::Handle<reco::Centrality> cent;
+  
+
+  if (isCaloJet) mEvent.getByToken(caloJetsToken_, caloJets);
+  if (isJPTJet)  mEvent.getByToken(jptJetsToken_, jptJets);
+  if (isPFJet) {  
+    if(std::string("Pu")==UEAlgo) mEvent.getByToken(basicJetsToken_, basicJets);
+    if(std::string("Vs")==UEAlgo) mEvent.getByToken(pfJetsToken_, pfJets);
+  }
+  
+  mEvent.getByToken(pfCandToken_, pfCandidates);
+  mEvent.getByToken(pfCandViewToken_, pfcandidates_);
+  
+  mEvent.getByToken(caloTowersToken_, caloCandidates);
+  mEvent.getByToken(caloCandViewToken_, calocandidates_);
+
+  mEvent.getByToken(backgrounds_, VsBackgrounds);
+  mEvent.getByToken(backgrounds_value_, vn_);
+
+  // get the centrality 
+  mEvent.getByToken(centralityToken_, cent);
+
+  mHF->Fill(cent->EtHFtowerSum());
+  Float_t HF_energy = cent->EtHFtowerSum();
+
+  const reco::PFCandidateCollection *pfCandidateColl = pfCandidates.product();
+  
+  Float_t vsPt=0;
+  Float_t vsPtInitial = 0;
+  Float_t vsArea = 0;
+  Int_t NPFpart = 0;
+  Int_t NCaloTower = 0;
+  Float_t pfPt = 0;
+  Float_t pfEta = 0;
+  Float_t pfPhi = 0;
+  Float_t caloPt = 0;
+  Float_t caloEta = 0;
+  Float_t caloPhi = 0;
+  Float_t SumPt_value = 0;
+  Float_t vn_value[200];
+  Float_t psin_value[200];
+
+  Float_t vn[fourierOrder_][etaBins_];
+  Float_t psin[fourierOrder_][etaBins_];
+  Float_t sumpT[etaBins_];
+
+  double edge_pseudorapidity[etaBins_ +1] = {-5.191, -2.650, -2.043, -1.740, -1.479, -1.131, -0.783, -0.522, 0.522, 0.783, 1.131, 1.479, 1.740, 2.043, 2.650, 5.191 };
+  
+  UEParameters vnUE(vn_.product(),fourierOrder_,etaBins_);
+  const std::vector<float>& vue = vnUE.get_raw();
```

Comment:
ciao @ktf - why do we see this problem only gcc481? do you know?

On Feb 6, 2015, at 10:55 AM, Giulio Eulisse <notifications@github.com>
 wrote:

> In DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc:
> 
> > +  Float_t pfPhi = 0;
> > +  Float_t caloPt = 0;
> > +  Float_t caloEta = 0;
> > +  Float_t caloPhi = 0;
> > +  Float_t SumPt_value = 0;
> > +  Float_t vn_value[200];
> > +  Float_t psin_value[200];
> > +
> > +  Float_t vn[fourierOrder_][etaBins_];
> > +  Float_t psin[fourierOrder_][etaBins_];
> > +  Float_t sumpT[etaBins_];
> > +
> > +  double edge_pseudorapidity[etaBins_ +1] = {-5.191, -2.650, -2.043, -1.740, -1.479, -1.131, -0.783, -0.522, 0.522, 0.783, 1.131, 1.479, 1.740, 2.043, 2.650, 5.191 };
> > +  
> > +  UEParameters vnUE(vn_.product(),fourierOrder_,etaBins_);
> > +  const std::vector<float>& vue = vnUE.get_raw();
> 
> This is breaking the gcc481 IB due to the fact it's unused. Can you please fix it ASAP? @deguio
> 
> �
> Reply to this email directly or view it on GitHub.
>